src/frontegg-oauth-client.ts

```diff
@@ -391,31 +401,29 @@ export class FronteggOAuthClient {
   }
 
   /**
-   * Function to fetch the user details from Frontegg.
-   * If the user is not authenticated or the access token is expired, the function throws error.
-   * More information: https://docs.frontegg.com/reference/userscontrollerv2_getuserprofile
+   * Function to decode the user JWT access token and extract the user data.
    */
-  private async fetchUserData(userAccessToken: string): Promise<FronteggUserData> {
-    const response = await fetchWithAssert(
-      `${this.baseUrl}/frontegg/identity/resources/users/v2/me`,
-      {
-        credentials: 'include',
-        headers: {
-          Authorization: `Bearer ${userAccessToken}`,
-          'Content-Type': 'application/json',
-        },
-      },
+  private decodeAccessToken(userAccessToken: string) {
+    const base64Url = userAccessToken.split('.')[1]
+    const base64 = base64Url.replace(/-/g, '+').replace(/_/g, '/')
+    const jsonPayload = decodeURIComponent(
+      window
+        .atob(base64)
+        .split('')
```

Comment:
🟡 Is this necessary? Shouldn't just `atob(base64)` do the job?

Reply:
`atob` apparently doesn't properly decode utf8: https://stackoverflow.com/a/30106551. But I think it might actually be worth just using a library for the decoding: https://github.com/auth0/jwt-decode/blob/main/lib/index.ts. It's very lightweight but adds a couple of additional checks.

Reply:
That's very important piece of the puzzle! :) 

I'm more in favour of copying over only what we use (then installing dependency) and maybe abstract it into a function with a comment that explains it all (why we don't use native base64 encoding, etc.).

Reply:
Apologies for radio silence, we had to wait for Frontegg to fix a bug with the impersonation object (`act` object) returned in their JWT response. That's now resolved. I have updated the PR with:
- separate function to decode a JWT, with documentation
- additional `impersonated: boolean` field on the user data response
- additional test coverage for impersonated and regular sessions